resources/migrations/001_update_migrations.yaml

```diff
@@ -9383,6 +9383,242 @@ databaseChangeLog:
+                    referencedColumnNames: id
+                    foreignKeyName: fk_notification_handler_notification_id
+                    deleteCascade: true
+              - column:
+                  name: channel_id
+                  type: int
+                  remarks: the channel that the handler is connected to
+                  constraints:
+                    nullable: true # temporarily nullable, because we need to support email and slack channels as global configuration
+                    referencedTableName: channel
+                    referencedColumnNames: id
+                    foreignKeyName: fk_notification_handler_channel_id
+                    deleteCascade: true
+              - column:
+                  name: active
+                  type: ${boolean.type}
+                  remarks: whether the handler is active
+                  defaultValueBoolean: true
+                  constraints:
+                    nullable: false
+              - column:
+                  name: created_at
+                  type: ${timestamp_type}
+                  remarks: The timestamp of when the handler was created
+                  defaultValueComputed: current_timestamp
+                  constraints:
+                    nullable: false
+              - column:
+                  name: updated_at
+                  type: ${timestamp_type}
+                  remarks: The timestamp of when the handler was updated
+                  defaultValueComputed: current_timestamp
+                  constraints:
+                    nullable: false
+
+  - changeSet:
+      id: v51.2024-09-05T08:00:05
+      author: qnkhuat
+      comment: index on notification_handler.notification_id
+      rollback: # not necessary, will be removed with the table
+      preConditions:
+        - not:
+            - indexExists:
+                tableName: notification_handler
+                indexName: idx_notification_handler_notification_id
+      changes:
+        - createIndex:
+            tableName: notification_handler
+            columns:
+              - column:
+                  name: notification_id
+            indexName: idx_notification_handler_notification_id
+
+  - changeSet:
+      id: v51.2024-09-05T08:00:06
+      author: qnkhuat
+      comment: index on notification_handler.channel_id
+      rollback: # not necessary, will be removed with the table
+      preConditions:
+        - not:
+            - indexExists:
+                tableName: notification_handler
+                indexName: idx_notification_handler_channel_id
+      changes:
+        - createIndex:
+            tableName: notification_handler
+            columns:
+              - column:
+                  name: channel_id
+            indexName: idx_notification_handler_channel_id
+
+  - changeSet:
+      id: v51.2024-09-05T08:00:07
+      author: qnkhuat
+      comment: Create the notification_recipient table
```

Comment:
Will we not need to support external recipients? (i.e. a plain email address)

Reply:
yes, in that case we'll have 
```clojure
{:type :notification-recipient/external-email
 :details "chris@metabase.com"}
```

`details` is different for each type and used for custom ones

Reply:
Ah ok. What do you think about `:configuration` instead? Details sounds non-functional.

Reply:
yea configuration does sound better, I'll update it in the upstream PR

Reply:
on JSON, yes!